### PR TITLE
perf.py: Added perf get events

### PR DESCRIPTION
**Pull request by nasastry**
get_events -- this function returns only the perf events matching with
supplied pattern

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by clebergnu** [comment]
I think this could be introduced as an `avocado.utils.process` utility instead, one that takes a command and a pattern.  I don't think this justifies a new module (`avocado.utils.perf`) and the hardcoded `perf list` list command.  If, the `perf list` command output were to be parsed into a proper data structure, then I think it'd make sense to have a perf module.

**Comment by nasastry** [comment]
@clebergnu Added some more functions related to perf events. Please review. Thank you.

File: avocado/utils/perf.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    :rtype: list of str.
+    """
+    list_of_events = []
+    for line in process.run("perf list").stdout.split('\n'):
+        if pattern in line:
+            list_of_events.append(line)
+    return list_of_events
+
+
+def get_hv_gpci_events():
```
Comment (PraveenPenguin) [inline avocado/utils/perf.py]: i think good option is we depend on user to just use `get_events` with `pattern` as decide in test case what even need to used

File: avocado/utils/process.py [hunk 2]
```diff
@@ -1641,3 +1641,21 @@ def get_owner_id(pid):
...
+    list and return it.
+
+    :param pattern: Pattern to search.
+    :type pattern: str.
+
+    :return: list of events matching the 'pattern'.
+    :rtype: list of str.
+    """
+    list_of_events = []
+    for line in run("perf list").stdout.split('\n'):
```
Comment (PraveenPenguin) [inline avocado/utils/process.py]: using `system_output` function will good as it already return `stdout` of command

Comment (PraveenPenguin) [inline avocado/utils/process.py]: using `system_output` function will good as it already return `stdout` of command

Comment (nasastry) [reply avocado/utils/process.py]: @PraveenPenguin Thanks for your feedback.
To get the required output I have to do the following,

`with run`
```
for line in process.run("perf list").stdout.split('\n'):
```

`with system_output`
```
for line in process.system_output("perf list").split('\n'):
```

I don't see much difference between those two call outs.

Comment (PraveenPenguin) [reply avocado/utils/process.py]: @nasastry yes both will does some that's why saying `system_output` can be used

**Comment by nasastry** [comment]
@PraveenPenguin get_perf_events() in process.py updated with the required `system_output`

**Pull request merged**